addons/web/static/src/views/fields/text/text_field.js

```diff
@@ -18,6 +22,43 @@ export class TextField extends Component {
                 this.resize();
             }
         });
+        onMounted(this.onMounted);
+        onWillUnmount(this.onWillUnmount);
+    }
+    async onKeydownListener(ev) {
+        if (ev.key === this.dynamicPlaceholder.TRIGGER_KEY && ev.target === this.textareaRef.el) {
+            const baseModel = this.props.record.data.mailing_model_real;
+            if (baseModel) {
+                await this.dynamicPlaceholder.open(
+                    this.textareaRef.el,
+                    baseModel,
+                    this.dynamicplaceholderValidate.bind(this),
+                    this.dynamicplaceholderClose.bind(this)
+                );
+            }
+        }
+    }
+    onMounted() {
+        if (this.props.dynamicPlaceholder) {
+            this.keydownListenerCallback = this.onKeydownListener.bind(this);
+            document.addEventListener('keydown', this.keydownListenerCallback);
+        }
+    }
+    onWillUnmount() {
+        if (this.props.dynamicPlaceholder) {
+            document.removeEventListener('keydown', this.keydownListenerCallback);
+        }
+    }
+    dynamicplaceholderValidate(chain, defaultValue) {
+        if (chain) {
+            const triggerKeyReplaceRegex = new RegExp(`${this.dynamicPlaceholder.TRIGGER_KEY}$`);
+            let dynamicPlaceholder = "{{object." + chain.join('.');
+            dynamicPlaceholder += defaultValue && defaultValue !== '' ? ` or '''${defaultValue}'''}}` : '}}';
+            this.textareaRef.el.value = this.textareaRef.el.value.replace(triggerKeyReplaceRegex, '') + dynamicPlaceholder;
+        }
+    }
+    dynamicplaceholderClose() {
```

Comment:
No capital letter for placeholder ?

Reply:
Also, shouldn't these be called more like "onDynamicPlaceholderClose" ? Makes more sense to me but I might be missing something ?